web/client-api/src/main/java/io/deephaven/web/client/api/filter/FilterValue.java

```diff
@@ -45,20 +50,54 @@ public static FilterValue ofString(@TsTypeRef(Any.class) Object input) {
         return new FilterValue(lit);
     }
 
+    @TsUnion
+    @JsType(name = "?", namespace = JsPackage.GLOBAL, isNative = true)
+    public interface OfNumberUnionParam {
+        @JsOverlay
+        static OfNumberUnionParam of(@DoNotAutobox Object value) {
+            return Js.cast(value);
+        }
+
+        @JsOverlay
+        default boolean isLongWrapper() {
+            return this instanceof LongWrapper;
+        }
+
+        @JsOverlay
+        default boolean isNumber() {
+            return (Object) this instanceof Double;
+        }
+
+        @TsUnionMember
+        @JsOverlay
+        default LongWrapper asLongWrapper() {
+            return Js.cast(this);
+        }
+
+        @TsUnionMember
+        @JsOverlay
+        default double asNumber() {
+            return Js.asDouble(this);
+        }
+    }
+
     @JsMethod(namespace = "dh.FilterValue")
-    public static FilterValue ofNumber(@TsTypeRef(Any.class) Object input) {
+    public static FilterValue ofNumber(OfNumberUnionParam input) {
         Objects.requireNonNull(input);
-        if (input instanceof DateWrapper) {
-            Literal lit = new Literal();
-            lit.setNanoTimeValue(((DateWrapper) input).valueOf());
-            return new FilterValue(lit);
-        } else if (input instanceof LongWrapper) {
-            Literal lit = new Literal();
-            lit.setLongValue(((LongWrapper) input).valueOf());
-            return new FilterValue(lit);
-        } else if (Js.typeof(input).equals("number")) {
+        if (input.isLongWrapper()) {
+            LongWrapper value = input.asLongWrapper();
+            if (value instanceof DateWrapper) {
+                Literal lit = new Literal();
+                lit.setNanoTimeValue(((DateWrapper) input).valueOf());
+                return new FilterValue(lit);
+            } else {
+                Literal lit = new Literal();
+                lit.setLongValue(((LongWrapper) input).valueOf());
+                return new FilterValue(lit);
+            }
+        } else if (input.isNumber()) {
             Literal lit = new Literal();
-            lit.setDoubleValue(Js.asDouble(input));
+            lit.setDoubleValue(input.asNumber());
             return new FilterValue(lit);
         } else {
             // not sure what the input is, try to toString(), then parse to Double, and use that
```

Comment:
Technically this supports any non-null Object in JS, but I didn't union that in, under the assumption (mistaken?) that while plain JS might well make that mistake, a TS project isn't going to pass an already-formatted string to an api that is expected to parse and then format it again...?

Reply:
I'll need to double check the web UI, there seemed to be cases where we passing a string. Also see the BigDecimal/BigInteger question.

Reply:
Ok, I'll add in string and Object, so that anything can work? I sort of question using this on strings to make it back into a string again though, unless this just "mostly works" for bigdecimal/integer

Reply:
Okay well just want to note that we may need to update this when fixing BigDecimal/BigInteger: https://github.com/deephaven/web-client-ui/issues/1216

Right now Web UI is just passing in the value that it gets from the API (when you right-click on a cell and select `Filter -> is exactly` or something): 
https://github.com/deephaven/web-client-ui/blob/26011467be3bb5947a2cf34d78eaaaedc47d909b/packages/jsapi-utils/src/TableUtils.ts#L1679